src/app/modules/layout/navbar/navbar.component.html

```diff
@@ -1,25 +1,32 @@
 <div>
     <span>
-      <img *ngIf="isDarkTheme" src="../../../../assets/images/logo.png" width="120px" alt="Clean CaDET" [routerLink]="['home']" (click)="resetNavBar()">
-      <img *ngIf="!isDarkTheme" src="../../../../assets/images/black.png" width="120px" alt="Clean CaDET" [routerLink]="['home'] " (click)="resetNavBar()">
+      <img *ngIf="isDarkTheme" src="../../../../assets/images/logo.png" width="120px" alt="Clean CaDET"
+           [routerLink]="['home']" (click)="resetNavBar()">
+      <img *ngIf="!isDarkTheme" src="../../../../assets/images/black.png" width="120px" alt="Clean CaDET"
+           [routerLink]="['home'] " (click)="resetNavBar()">
     </span>
-    <button color="primary" id="unit" mat-flat-button [disabled]="!learner" [matMenuTriggerFor]="unit">
-      {{selectedUnit ? selectedUnit.name : 'Select Unit'}}
+  <button color="primary" id="unit" mat-flat-button [disabled]="!learner" [matMenuTriggerFor]="unit">
+    {{selectedUnit ? selectedUnit.name : 'Select Unit'}}
+  </button>
+  <mat-menu #unit="matMenu">
+    <button (click)="onUnitSelected(unit)" [routerLink]="['unit/' + unit.id]" *ngFor="let unit of units" mat-menu-item>
+      {{unit.name}}
     </button>
-    <mat-menu #unit="matMenu">
-      <button (click)="onUnitSelected(unit)" [routerLink]="['unit/' + unit.id]" *ngFor="let unit of units" mat-menu-item>
-        {{unit.name}}
-      </button>
-    </mat-menu>
-    <button color="primary" id="kc" mat-flat-button [matMenuTriggerFor]="kc" [disabled]="selectedUnit == null">
-      {{selectedKC ? selectedKC.name : 'Select Knowledge Component'}}
-    </button>
-    <mat-menu #kc="matMenu">
-      <button (click)="onKCSelected(kc)" [routerLink]="['kc/' + kc.id]" *ngFor="let kc of knowledgeComponents" mat-menu-item>
+  </mat-menu>
+  <button color="primary" id="kc" mat-flat-button [matMenuTriggerFor]="kc" [disabled]="selectedUnit == null">
+    {{selectedKC ? selectedKC.name : 'Select Knowledge Component'}}
+  </button>
+  <mat-menu #kc="matMenu">
+    <ng-container *ngIf="selectedUnit != null">
+      <button (click)="onKCSelected(kc)" [routerLink]="['kc/' + kc.id + '/learner/' + learner.id]"
+              *ngFor="let kc of knowledgeComponents" mat-menu-item>
         {{kc.name}}
       </button>
-    </mat-menu>
-    <button color="primary" *ngIf="learner" id="profile" mat-flat-button [routerLink]="['profile']">Profile</button>
-    <button color="primary" *ngIf="!learner" id="login" mat-flat-button [routerLink]="['login']">Login</button>
-    <button color="primary" *ngIf="learner" mat-flat-button id="logout" (click)="onLogout()" [routerLink]="['home']">Logout</button>
+    </ng-container>
+  </mat-menu>
+  <button color="primary" *ngIf="learner" id="profile" mat-flat-button [routerLink]="['profile']">Profile</button>
+  <button color="primary" *ngIf="!learner" id="login" mat-flat-button [routerLink]="['login']">Login</button>
```

Comment:
Are these changes only stylistic and enforced by your IDE? If so, we should setup a common convention to avoid always overriding each other's styles.

Please send me your configuration if you've already established some conventions following best practices. If not, revert the changes and disable any automatic styling while we decide on the common set of conventions in the coming months.

Reply:
No, there was problem with rebase. There were two files under name ''navbar.component.html" so i removed one and moved other to modules/layout/navbar/.

Reply:
Reverted.